packages/devtools_app/lib/src/charts/chart.dart

```diff
@@ -7,11 +7,26 @@ import 'dart:ui';
 
 import 'package:flutter/cupertino.dart';
 import 'package:flutter/material.dart';
+import 'package:intl/intl.dart' as intl;
 
 import '../auto_dispose_mixin.dart';
 import 'chart_controller.dart';
 import 'chart_trace.dart';
 
+/// Helper function to:
+///   1. save canvas state
+///   2. translate coordinates
+///   3. draw to the canvas with respect to translation coordinates
+///   4. restore canvas state back to the saved state.
+void translate(Canvas canvas, double x, double y, VoidCallback drawCode) {
```

Comment:
maybe make 
drawCode take the `canvas` as a parameter rather than being a void callback?

Reply:
typedef DrawCodeCallback = void Function(Canvas canvas);
void translate(Canvas canvas, double x, double y, DrawCodeCallback drawCode) 

Done.